src/http-service.js

```diff
@@ -0,0 +1,78 @@
+import 'whatwg-fetch';
+
+export const http = {
+    baseSettings: {
+        credentials: 'same-origin'
+    },
+    getHtml: function (url) {
+        return fetch(url, this.baseSettings)
+            .then((response) => response.text());
+    },
+    getJSON: function (url) {
+        return fetch(url, this.baseSettings)
+            .then((response) => response.json());
+    },
+
+    post: function (url, body, extraHeaders) {
+        const checkStatus = function (response) {
+            if (response.status >= 400) {
+                return Promise.reject(response);
+            }
+
+            return Promise.resolve(response);
+        };
+
+        const plainHeaders = {
+            'X-Requested-With': 'XMLHttpRequest'
+        };
+
+        if (extraHeaders) {
+            Object.assign(plainHeaders, extraHeaders);
+        }
+
+        const headers = new Headers(plainHeaders);
+
+        const options = Object.assign({
+            method: 'POST',
+            body,
+            headers
+        }, this.baseSettings);
+
+        return fetch(url, options)
+            .then(checkStatus);
+    },
+
+    getCsrfToken: function () {
+        const csrfCookieRegex = /.*csrftoken=(\w+).*/;
+        const csrfToken = document.cookie.match(csrfCookieRegex)[1];
+
+        return csrfToken;
+    },
+
+    postDataAsForm: function (url, data) {
+        const formData = new FormData();
```

Comment:
The `body` of a fetch can be one of:
>body: Any body that you want to add to your request: this can be a Blob, BufferSource, FormData, URLSearchParams, or USVString object. Note that a request using the GET or HEAD method cannot have a body.
Or just a JSON string.